controllers/rack.go

```diff
@@ -785,39 +804,42 @@ func (r *SingleClusterReconciler) scaleDownRack(
 		)
 	}
 
-	// Wait for pods to get terminated
-	if err = r.waitForSTSToBeReady(found); err != nil {
-		r.Log.Error(err, "Failed to wait for statefulset to be ready")
-		return found, reconcileRequeueAfter(1)
-	}
-
-	// This check is added only in scale down but not in rolling restart.
-	// If scale down leads to unavailable or dead partition then we should scale up the cluster,
-	// This can be left to the user but if we would do it here on our own then we can reuse
-	// objects like pvc and service. These objects would have been removed if scaleup is left for the user.
-	// In case of rolling restart, no pod cleanup happens, therefor rolling config back is left to the user.
-	if err = r.validateSCClusterState(policy, ignorablePods); err != nil {
-		// reset cluster size
-		newSize := *found.Spec.Replicas + 1
-		found.Spec.Replicas = &newSize
+	// No need for these checks if pod is not running. These checks will fail if there is any other pod in failed state.
+	if isPodRunningAndReady {
```

Comment:
should we get the running and ready status again, instead of using the old variable.